source/write/bulk-write.txt

```diff
@@ -0,0 +1,287 @@
+
+The following sections show how to create and use operation classes for
+different write operations.
+
+Insert Operations
+~~~~~~~~~~~~~~~~~
+
+To perform an insert operation, create an instance of ``InsertOne`` and specify
+the document you want to insert. To insert multiple documents, you must create
+an instance of ``InsertOne`` for each document.
+
+The following example creates an instance of ``InsertOne``:
+
+.. literalinclude:: /includes/write/bulk-write.py
+   :start-after: start-bulk-insert-one
+   :end-before: end-bulk-insert-one
+   :language: python
+   :copyable:
+
+Update Operations
+~~~~~~~~~~~~~~~~~
+
+To perform a bulk update operation, create an instance of ``UpdateOne`` or
+``UpdateMany`` and pass it a **query filter** and the update operation you want
+to perform. A query filter specifies the criteria used to match documents to
+in your collection.
+
+``UpdateOne`` updates *the first* document that matches your query filter.
+``UpdateMany`` updates *all* documents that match your query filter.
+
+The following example creates an instance of ``UpdateOne``:
+
+.. literalinclude:: /includes/write/bulk-write.py
+   :start-after: start-bulk-update-one
+   :end-before: end-bulk-update-one
+   :language: python
+   :copyable:
+
+The following example creates an instance of ``UpdateMany``:
+
+.. literalinclude:: /includes/write/bulk-write.py
+   :start-after: start-bulk-update-many
+   :end-before: end-bulk-update-many
+   :language: python
+   :copyable:
+
+Replace Operations
+~~~~~~~~~~~~~~~~~~
+
+Replace operations remove all fields and values of a specified document and
+replaces them with new ones. To perform a replace operation, create an instance
+of ``ReplaceOne`` and pass it a query filter and the fields and values you want
+to replace with. To replace multiple documents, you must create
+an instance of ``ReplaceOne`` for each document.
+
+The following example creates an instance of ``ReplaceOne``:
+
+.. literalinclude:: /includes/write/bulk-write.py
+   :start-after: start-bulk-replace-one
+   :end-before: end-bulk-replace-one
+   :language: python
+   :copyable:
+
+Delete Operations
+~~~~~~~~~~~~~~~~~
+
+To perform a delete operation, create an instance of ``DeleteOne`` or ``DeleteMany`` and pass it a
+query filter specifying the documents you want to delete.
+
+``DeleteOne`` removes *the first* document that matches your query filter.
+``DeleteMany`` removes *all* documents that match your query filter.
+
+The following example creates an instance of ``DeleteOne``:
+
+.. literalinclude:: /includes/write/bulk-write.py
+   :start-after: start-bulk-delete-one
+   :end-before: end-bulk-delete-one
+   :language: python
+   :copyable:
+
+The following example creates an instance of ``DeleteMany``:
+
+.. literalinclude:: /includes/write/bulk-write.py
+   :start-after: start-bulk-delete-many
+   :end-before: end-bulk-delete-many
+   :language: python
+   :copyable:
+
+Mixed Bulk Write Operations
+----------------------------
```

Comment:
This section is the "main example" of how to perform a bulk write. I put it farther down the page because you need to use the operations described in the earlier sections to be able to use it. However, this puts the example 3/4 down the page, which users have complained about in the past. Should we move this section up and note that it's explained later? Or maybe add another example in the overview? Open to suggestions here.

Reply:
Since we're separating usage examples by topic and putting them right up front, I think it's okay if longer code examples are farther down the page in "Fundamentals."